File: LeetCode/Easy/Split a String in Balanced Strings/1221.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function(s) {
    let cnt = 0;

    let balance = 0;
    for (let i = 0; i < s.length; i++) {
        balance += (s[i] === 'L') ? 1 : -1;
        if (balance === 0) cnt++;
    }

    return cnt;
};

File: LeetCode/Easy/Split a String in Balanced Strings/1221.java
class Solution {
    public int balancedStringSplit(String s) {
        int cnt = 0;

        int balance = 0;
        for (int i = 0; i < s.length(); i++) {
            balance += (s.charAt(i) == 'L') ? 1 : -1;
            if (balance == 0) cnt++;
        }

        return cnt;
    }
}
